uploadgui/src/main/scala/se/lu/nateko/cp/meta/upload/subforms/DocumentPanel.scala

```diff
@@ -0,0 +1,72 @@
+package se.lu.nateko.cp.meta.upload.subforms
+
+import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
+
+import se.lu.nateko.cp.meta.upload._
+import se.lu.nateko.cp.meta.{UploadDto, DocObjectDto}
+import se.lu.nateko.cp.meta.core.data.Envri
+
+import formcomponents._
+
+class DocumentPanel(implicit bus: PubSubBus, envri: Envri.Envri) extends PanelSubform(".document-section"){
+	def title = documentTitle.value
+	def description = documentDescription.value
+	def authors = documentAuthors.values
+
+	private val documentTitle = new TextOptInput("document-title", notifyUpdate)
+	private val documentDescription = new TextOptInput("document-description", notifyUpdate)
+	private val agentList = new DataList[NamedUri]("agent-list", _.name)
+	private val documentAuthors = new DataListForm("document-authors", agentList, notifyUpdate)
+	private val agentAgg = new AgentAggregator
+
+	def resetForm(): Unit = {
+		documentTitle.reset()
+		documentDescription.reset()
+		documentAuthors.setValues(Seq())
+	}
+
+	override def show(): Unit = {
+		super.show()
+		for(
```

Comment:
It seems like the following parts of code have been copy-pasted from Production Panel. Can we come up with some way of code reuse instead?